src/core/emergency-action.c

```diff
@@ -28,10 +40,22 @@ void emergency_action(
                 int exit_status,
                 const char *reason) {
 
+        Unit *u;
+
         assert(m);
         assert(action >= 0);
         assert(action < _EMERGENCY_ACTION_MAX);
 
+        /* Is the special shutdown target active or queued? If so, we are in shutdown state */
+        u = manager_get_unit(m, SPECIAL_SHUTDOWN_TARGET);
+        if (u && unit_active_or_pending(u) &&
+            IN_SET(action, EMERGENCY_ACTION_REBOOT,
+                   EMERGENCY_ACTION_POWEROFF, EMERGENCY_ACTION_EXIT)) {
+                log_notice("EmergencyAction: Shutdown is already active Skipping %s request",
+                         emergency_action_table[action]);
```

Comment:
Indentation, and the sentences run together. Maybe "Shutdown is already active. Skipping emergency action request %s."